### ISSUE TITLE
The deployment deployed in karmada is always in the Not Ready state.

### ISSUE THREAD

**Issue by hansedong** [karmada-io/karmada#397]
**What happened**:

I created 2 Clusters, and they are all normal.

```
[root@karmada-node1 operation]# kubectl get clusters
NAME   VERSION   MODE   READY   AGE
boat   v1.21.1   Push   True    4h20m
hull   v1.13.9   Push   True    126m
```
I deployed demo Deployment in the karmada cluster, and it was also normally scheduled to the target K8s cluster.
```
[root@karmada-node1 operation]# cat nginx.deploy.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
  labels:
    app: nginx
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - image: nginx
        name: nginx
```
```
[root@karmada-node1 operation]# cat propagationpolicy.yaml
apiVersion: policy.karmada.io/v1alpha1
kind: PropagationPolicy
metadata:
  name: nginx-propagation
spec:
  resourceSelectors:
    - apiVersion: apps/v1
      kind: Deployment
      name: nginx
      namespace: default
  placement:
    clusterAffinity:
      clusterNames:
        - boat
```
After deploying Nginx Deployment, in the K8s cluster (named boat), you can see that the Pod is created successfully.
```
[root@k8s-boat-node1 operation]# kubectl get po -o wide
NAME                     READY   STATUS    RESTARTS   AGE     IP               NODE                  NOMINATED NODE   READINESS GATES
nginx-6799fc88d8-7pvq9   1/1     Running   0          29m     172.16.197.244   knode172-16-130-152   <none>           <none>
```
However, the Nginx Deployment status seen in the karmada cluster is abnormal ( the Ready state is 0/1 ).
```
[root@karmada-node1 operation]# kubectl get deploy
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   0/1     0            0           32m
```

**What you expected to happen**:

The Ready state of Nginx Deployment is 1/1.

**How to reproduce it (as minimally and precisely as possible)**:

I have no idea...

**Anything else we need to know?**:

1. karmada controller-manager keeps refreshing the log

```
I0601 13:57:23.043449   75716 binding_controller.go:42] Reconciling ResourceBinding default/nginx-deployment.
I0601 13:57:23.049472   75716 detector.go:891] Reconciling resource binding(default/nginx-deployment)
I0601 13:57:23.049545   75716 detector.go:989] Scrub deployment(default/nginx) status from cluster(boat), replicas: 1, ready: 1, updated: 1, available: 1, unavailable: 0
I0601 13:57:23.075362   75716 binding.go:278] Update work karmada-es-boat/default-nginx-deployment successfully.
I0601 13:57:23.075851   75716 workstatus.go:31] New aggregatedStatuses are equal with old resourceBinding(default/nginx-deployment) AggregatedStatus, no update required.
I0601 13:57:23.222473   75716 detector.go:149] Reconciling object: {apps v1 Deployment default nginx}
I0601 13:57:23.222623   75716 detector.go:259] attempts to match policy for resource({apps v1 Deployment default nginx})
I0601 13:57:23.222721   75716 detector.go:289] Matched policy(default/nginx-propagation) for resource({apps v1 Deployment default nginx})
I0601 13:57:23.222746   75716 detector.go:331] Applying policy(nginx-propagation) for object: {apps v1 Deployment default nginx}
I0601 13:57:23.236874   75716 detector.go:360] Update ResourceBinding(default/nginx-deployment) successfully.
I0601 13:57:23.237437   75716 binding_controller.go:42] Reconciling ResourceBinding default/nginx-deployment.
I0601 13:57:23.243017   75716 detector.go:891] Reconciling resource binding(default/nginx-deployment)
I0601 13:57:23.243099   75716 detector.go:989] Scrub deployment(default/nginx) status from cluster(boat), replicas: 1, ready: 1, updated: 1, available: 1, unavailable: 0
I0601 13:57:23.262515   75716 binding.go:278] Update work karmada-es-boat/default-nginx-deployment successfully.
I0601 13:57:23.262756   75716 workstatus.go:31] New aggregatedStatuses are equal with old resourceBinding(default/nginx-deployment) AggregatedStatus, no update required.
I0601 13:57:23.422236   75716 detector.go:149] Reconciling object: {apps v1 Deployment default nginx}
I0601 13:57:23.422490   75716 detector.go:259] attempts to match policy for resource({apps v1 Deployment default nginx})
I0601 13:57:23.422546   75716 detector.go:289] Matched policy(default/nginx-propagation) for resource({apps v1 Deployment default nginx})
I0601 13:57:23.422556   75716 detector.go:331] Applying policy(nginx-propagation) for object: {apps v1 Deployment default nginx}
I0601 13:57:23.440020   75716 binding_controller.go:42] Reconciling ResourceBinding default/nginx-deployment.
I0601 13:57:23.440578   75716 detector.go:360] Update ResourceBinding(default/nginx-deployment) successfully.
I0601 13:57:23.444493   75716 detector.go:891] Reconciling resource binding(default/nginx-deployment)
I0601 13:57:23.444612   75716 detector.go:989] Scrub deployment(default/nginx) status from cluster(boat), replicas: 1, ready: 1, updated: 1, available: 1, unavailable: 0
I0601 13:57:23.471441   75716 binding.go:278] Update work karmada-es-boat/default-nginx-deployment successfully.
I0601 13:57:23.471740   75716 workstatus.go:31] New aggregatedStatuses are equal with old resourceBinding(default/nginx-deployment) AggregatedStatus, no update required.
I0601 13:57:23.623153   75716 detector.go:149] Reconciling object: {apps v1 Deployment default nginx}
I0601 13:57:23.623380   75716 detector.go:259] attempts to match policy for resource({apps v1 Deployment default nginx})
I0601 13:57:23.623720   75716 detector.go:289] Matched policy(default/nginx-propagation) for resource({apps v1 Deployment default nginx})
I0601 13:57:23.623752   75716 detector.go:331] Applying policy(nginx-propagation) for object: {apps v1 Deployment default nginx}
I0601 13:57:23.651208   75716 detector.go:360] Update ResourceBinding(default/nginx-deployment) successfully.
I0601 13:57:23.651683   75716 binding_controller.go:42] Reconciling ResourceBinding default/nginx-deployment.
I0601 13:57:23.656940   75716 detector.go:891] Reconciling resource binding(default/nginx-deployment)
I0601 13:57:23.657077   75716 detector.go:989] Scrub deployment(default/nginx) status from cluster(boat), replicas: 1, ready: 1, updated: 1, available: 1, unavailable: 0
I0601 13:57:23.680716   75716 binding.go:278] Update work karmada-es-boat/default-nginx-deployment successfully.
I0601 13:57:23.680906   75716 workstatus.go:31] New aggregatedStatuses are equal with old resourceBinding(default/nginx-deployment) AggregatedStatus, no update required.
I0601 13:57:23.823937   75716 detector.go:149] Reconciling object: {apps v1 Deployment default nginx}
I0601 13:57:23.824137   75716 detector.go:259] attempts to match policy for resource({apps v1 Deployment default nginx})
I0601 13:57:23.824250   75716 detector.go:289] Matched policy(default/nginx-propagation) for resource({apps v1 Deployment default nginx})
I0601 13:57:23.824279   75716 detector.go:331] Applying policy(nginx-propagation) for object: {apps v1 Deployment default nginx}
I0601 13:57:23.841441   75716 binding_controller.go:42] Reconciling ResourceBinding default/nginx-deployment.
I0601 13:57:23.841775   75716 detector.go:360] Update ResourceBinding(default/nginx-deployment) successfully.
I0601 13:57:23.847918   75716 detector.go:891] Reconciling resource binding(default/nginx-deployment)
I0601 13:57:23.848085   75716 detector.go:989] Scrub deployment(default/nginx) status from cluster(boat), replicas: 1, ready: 1, updated: 1, available: 1, unavailable: 0
I0601 13:57:23.873069   75716 binding.go:278] Update work karmada-es-boat/default-nginx-deployment successfully.
I0601 13:57:23.873333   75716 workstatus.go:31] New aggregatedStatuses are equal with old resourceBinding(default/nginx-deployment) AggregatedStatus, no update required.
I0601 13:57:24.024367   75716 detector.go:149] Reconciling object: {apps v1 Deployment default nginx}
I0601 13:57:24.024574   75716 detector.go:259] attempts to match policy for resource({apps v1 Deployment default nginx})
I0601 13:57:24.024640   75716 detector.go:289] Matched policy(default/nginx-propagation) for resource({apps v1 Deployment default nginx})
I0601 13:57:24.024655   75716 detector.go:331] Applying policy(nginx-propagation) for object: {apps v1 Deployment default nginx}
I0601 13:57:24.039570   75716 detector.go:360] Update ResourceBinding(default/nginx-deployment) successfully.
I0601 13:57:24.039679   75716 binding_controller.go:42] Reconciling ResourceBinding default/nginx-deployment.
I0601 13:57:24.045369   75716 detector.go:891] Reconciling resource binding(default/nginx-deployment)
I0601 13:57:24.045466   75716 detector.go:989] Scrub deployment(default/nginx) status from cluster(boat), replicas: 1, ready: 1, updated: 1, available: 1, unavailable: 0
I0601 13:57:24.077104   75716 binding.go:278] Update work karmada-es-boat/default-nginx-deployment successfully.
I0601 13:57:24.077412   75716 workstatus.go:31] New aggregatedStatuses are equal with old resourceBinding(default/nginx-deployment) AggregatedStatus, no update required.
I0601 13:57:24.220967   75716 detector.go:149] Reconciling object: {apps v1 Deployment default nginx}
I0601 13:57:24.221155   75716 detector.go:259] attempts to match policy for resource({apps v1 Deployment default nginx})
I0601 13:57:24.221307   75716 detector.go:289] Matched policy(default/nginx-propagation) for resource({apps v1 Deployment default nginx})
I0601 13:57:24.221324   75716 detector.go:331] Applying policy(nginx-propagation) for object: {apps v1 Deployment default nginx}
```
2. karmada webhook keeps refreshing the log

```
I0601 13:46:31.548448   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
I0601 13:46:31.738821   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
I0601 13:46:31.931904   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
I0601 13:46:32.130753   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
I0601 13:46:32.327714   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
I0601 13:46:32.529960   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
I0601 13:46:32.733402   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
I0601 13:46:32.928390   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
I0601 13:46:33.128090   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
I0601 13:46:33.336346   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
I0601 13:46:33.540768   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
I0601 13:46:33.726820   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
I0601 13:46:33.936525   22621 mutating.go:33] Mutating work(default-nginx-deployment) for request: UPDATE
```
3. the works.work.karmada.io CRD info

```
apiVersion: v1
items:
- apiVersion: work.karmada.io/v1alpha1
  kind: Work
  metadata:
    creationTimestamp: "2021-06-01T05:44:23Z"
    finalizers:
    - karmada.io/execution-controller
    generation: 1
    labels:
      resourcebinding.karmada.io/name: nginx-deployment
      resourcebinding.karmada.io/namespace: default
    name: default-nginx-deployment
    namespace: karmada-es-boat
    ownerReferences:
    - apiVersion: policy.karmada.io/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: ResourceBinding
      name: nginx-deployment
      uid: 1152a566-3e51-4c52-b486-f650ded1beca
    resourceVersion: "332568"
    uid: 26c0f84a-c7bd-45e9-b6e0-28e8274de26d
  spec:
    workload:
      manifests:
      - apiVersion: apps/v1
        kind: Deployment
        metadata:
          annotations:
            deployment.kubernetes.io/revision: "1"
            kubectl.kubernetes.io/last-applied-configuration: |
              {"apiVersion":"apps/v1","kind":"Deployment","metadata":{"annotations":{},"labels":{"app":"nginx"},"name":"nginx","namespace":"default"},"spec":{"replicas":1,"selector":{"matchLabels":{"app":"nginx"}},"template":{"metadata":{"labels":{"app":"nginx"}},"spec":{"containers":[{"image":"nginx","name":"nginx"}]}}}}
          labels:
            app: nginx
            propagationpolicy.karmada.io/name: nginx-propagation
            propagationpolicy.karmada.io/namespace: default
            resourcebinding.karmada.io/name: nginx-deployment
            resourcebinding.karmada.io/namespace: default
            work.karmada.io/name: default-nginx-deployment
            work.karmada.io/namespace: karmada-es-boat
          name: nginx
          namespace: default
        spec:
          progressDeadlineSeconds: 600
          replicas: 1
          revisionHistoryLimit: 10
          selector:
            matchLabels:
              app: nginx
          strategy:
            rollingUpdate:
              maxSurge: 25%
              maxUnavailable: 25%
            type: RollingUpdate
          template:
            metadata:
              creationTimestamp: null
              labels:
                app: nginx
            spec:
              containers:
              - image: nginx
                imagePullPolicy: Always
                name: nginx
                resources: {}
                terminationMessagePath: /dev/termination-log
                terminationMessagePolicy: File
              dnsPolicy: ClusterFirst
              restartPolicy: Always
              schedulerName: default-scheduler
              securityContext: {}
              terminationGracePeriodSeconds: 30
  status:
    conditions:
    - lastTransitionTime: "2021-06-01T05:44:08Z"
      message: Manifest has been successfully applied
      reason: AppliedSuccessful
      status: "True"
      type: Applied
    manifestStatuses:
    - Status:
        availableReplicas: 1
        conditions:
        - lastTransitionTime: "2021-06-01T05:44:24Z"
          lastUpdateTime: "2021-06-01T05:44:24Z"
          message: Deployment has minimum availability.
          reason: MinimumReplicasAvailable
          status: "True"
          type: Available
        - lastTransitionTime: "2021-06-01T05:44:19Z"
          lastUpdateTime: "2021-06-01T05:44:24Z"
          message: ReplicaSet "nginx-6799fc88d8" has successfully progressed.
          reason: NewReplicaSetAvailable
          status: "True"
          type: Progressing
        observedGeneration: 1
        readyReplicas: 1
        replicas: 1
        updatedReplicas: 1
      identifier:
        group: apps
        kind: Deployment
        name: nginx
        namespace: default
        ordinal: 0
        resource: ""
        version: v1
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

4. the resourcebindings.work.karmada.io CRD info
```
apiVersion: work.karmada.io/v1alpha1
kind: ResourceBinding
metadata:
  annotations:
    policy.karmada.io/applied-placement: '{"clusterAffinity":{"clusterNames":["boat"]}}'
  creationTimestamp: "2021-06-01T05:44:23Z"
  generation: 4238
  labels:
    propagationpolicy.karmada.io/name: nginx-propagation
    propagationpolicy.karmada.io/namespace: default
  name: nginx-deployment
  namespace: default
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: Deployment
    name: nginx
    uid: e22743ee-d99e-43e1-878b-c198da06e49c
  resourceVersion: "348515"
  uid: 1152a566-3e51-4c52-b486-f650ded1beca
spec:
  clusters:
  - name: boat
  resource:
    apiVersion: apps/v1
    kind: Deployment
    name: nginx
    namespace: default
    resourceVersion: "346396"
status:
  aggregatedStatus:
  - Status:
      availableReplicas: 1
      conditions:
      - lastTransitionTime: "2021-06-01T05:44:24Z"
        lastUpdateTime: "2021-06-01T05:44:24Z"
        message: Deployment has minimum availability.
        reason: MinimumReplicasAvailable
        status: "True"
        type: Available
      - lastTransitionTime: "2021-06-01T05:44:19Z"
        lastUpdateTime: "2021-06-01T05:44:24Z"
        message: ReplicaSet "nginx-6799fc88d8" has successfully progressed.
        reason: NewReplicaSetAvailable
        status: "True"
        type: Progressing
      observedGeneration: 1
      readyReplicas: 1
      replicas: 1
      updatedReplicas: 1
    clusterName: boat
```
In addition, when I use the command `kubectl get resourcebindings.work.karmada.io nginx-deployment -o yaml -w` to watch the information of this resourcebindings.work.karmada.io CRD, I found that the resourceVersion and generation fields are fast Constantly increasing.

**Environment**:
- Karmada version: 0.5 and 0.6.
- The two K8s clusters version, boat: 1.12.1, hull: 1.13.9.

**Comment by hansedong** [comment]
Some additional content:
1. All components of the karmada cluster have been deployed in systemd mode, not started via pod.
2. There is no kubelet component service in the karmada cluster.

**Comment by mrlihanbo** [comment]
It seems like the status of the deployment has been collected to ResourceBinding. Could you please provide the status of the deployment using the command "kubectl get deploy -oyaml nginx"? Then, we can check if the status of the deployment is updated by karmada controller.

**Comment by hansedong** [comment]
> It seems like the status of the deployment has been collected to ResourceBinding. Could you please provide the status of the deployment using the command "kubectl get deploy -oyaml nginx"? Then, we can check if the status of the deployment is updated by karmada controller.

```
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "1"
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"apps/v1","kind":"Deployment","metadata":{"annotations":{},"labels":{"app":"nginx"},"name":"nginx","namespace":"default"},"spec":{"replicas":1,"selector":{"matchLabels":{"app":"nginx"}},"template":{"metadata":{"labels":{"app":"nginx"}},"spec":{"containers":[{"image":"nginx","name":"nginx"}]}}}}
  creationTimestamp: "2021-06-01T07:07:46Z"
  generation: 1
  labels:
    app: nginx
    propagationpolicy.karmada.io/name: nginx-propagation
    propagationpolicy.karmada.io/namespace: default
  name: nginx
  namespace: default
  resourceVersion: "400264"
  uid: c5eee151-d839-4f33-b69b-c44e7a115c1b
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: nginx
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: nginx
    spec:
      containers:
      - image: nginx
        imagePullPolicy: Always
        name: nginx
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
status:
  conditions:
  - lastTransitionTime: "2021-06-01T07:06:38Z"
    lastUpdateTime: "2021-06-01T07:06:38Z"
    message: Deployment does not have minimum availability.
    reason: MinimumReplicasUnavailable
    status: "False"
    type: Available
  - lastTransitionTime: "2021-06-01T07:16:43Z"
    lastUpdateTime: "2021-06-01T07:16:43Z"
    message: ReplicaSet "nginx-6799fc88d8" has timed out progressing.
    reason: ProgressDeadlineExceeded
    status: "False"
    type: Progressing
  observedGeneration: 1
  replicas: 1
  unavailableReplicas: 1
  updatedReplicas: 1
```
As above, it is the content of nginx Deployment. In addition, I found that its resourceVersion is also continuously increasing. Now, the value of the resourceVersion field has increased to 400264.

**Comment by mrlihanbo** [comment]
> > It seems like the status of the deployment has been collected to ResourceBinding. Could you please provide the status of the deployment using the command "kubectl get deploy -oyaml nginx"? Then, we can check if the status of the deployment is updated by karmada controller.
> 
> ```
> apiVersion: apps/v1
> kind: Deployment
> metadata:
>   annotations:
>     deployment.kubernetes.io/revision: "1"
>     kubectl.kubernetes.io/last-applied-configuration: |
>       {"apiVersion":"apps/v1","kind":"Deployment","metadata":{"annotations":{},"labels":{"app":"nginx"},"name":"nginx","namespace":"default"},"spec":{"replicas":1,"selector":{"matchLabels":{"app":"nginx"}},"template":{"metadata":{"labels":{"app":"nginx"}},"spec":{"containers":[{"image":"nginx","name":"nginx"}]}}}}
>   creationTimestamp: "2021-06-01T07:07:46Z"
>   generation: 1
>   labels:
>     app: nginx
>     propagationpolicy.karmada.io/name: nginx-propagation
>     propagationpolicy.karmada.io/namespace: default
>   name: nginx
>   namespace: default
>   resourceVersion: "400264"
>   uid: c5eee151-d839-4f33-b69b-c44e7a115c1b
> spec:
>   progressDeadlineSeconds: 600
>   replicas: 1
>   revisionHistoryLimit: 10
>   selector:
>     matchLabels:
>       app: nginx
>   strategy:
>     rollingUpdate:
>       maxSurge: 25%
>       maxUnavailable: 25%
>     type: RollingUpdate
>   template:
>     metadata:
>       creationTimestamp: null
>       labels:
>         app: nginx
>     spec:
>       containers:
>       - image: nginx
>         imagePullPolicy: Always
>         name: nginx
>         resources: {}
>         terminationMessagePath: /dev/termination-log
>         terminationMessagePolicy: File
>       dnsPolicy: ClusterFirst
>       restartPolicy: Always
>       schedulerName: default-scheduler
>       securityContext: {}
>       terminationGracePeriodSeconds: 30
> status:
>   conditions:
>   - lastTransitionTime: "2021-06-01T07:06:38Z"
>     lastUpdateTime: "2021-06-01T07:06:38Z"
>     message: Deployment does not have minimum availability.
>     reason: MinimumReplicasUnavailable
>     status: "False"
>     type: Available
>   - lastTransitionTime: "2021-06-01T07:16:43Z"
>     lastUpdateTime: "2021-06-01T07:16:43Z"
>     message: ReplicaSet "nginx-6799fc88d8" has timed out progressing.
>     reason: ProgressDeadlineExceeded
>     status: "False"
>     type: Progressing
>   observedGeneration: 1
>   replicas: 1
>   unavailableReplicas: 1
>   updatedReplicas: 1
> ```
> 
> As above, it is the content of nginx Deployment. In addition, I found that its resourceVersion is also continuously increasing. Now, the value of the resourceVersion field has increased to 400264.

Hello, It seems strange for the status of the nginx deployment. There should be no conditions. Do you deploy the karmada apiserver? It seems like you using the kube-apiserver as the karmada-apiserver?

**Comment by hansedong** [comment]
@mrlihanbo 

Thank you very much for your prompt reply. I feel that this problem will soon be solved.

I did use kube-apiserver as karmada-apiserver. For the entire karmada environment, I deployed it based on centos systemd, not based on K8s Pod deployment.

I actually recompiled the karmada binary (via the make command), and there is no karmada-apiserver (in the karmada/cmd directory in the source code, there is no karmada-apiserver), and in the karmada/artifacts/deploy/karmada-apiserver.yaml file In the image field, the value is k8s.gcr.io/kube-apiserver:v1.19.1.

I used to think that karmada-apiserver is kube-apiserver (so karmada can be compatible with K8s resources). If not, is there any difference between kube-apiserver and karmada-apiserver? I want to deploy kube-apiserver in binary mode. What should I do?

**Comment by hansedong** [comment]
@mrlihanbo 

I watched nginx deploy in the karmada environment and found that in fact, the status status changes all the time, constantly changing to 0 and 1.

```
[root@karmada-node1 sample]# kubectl get deploy -w
NAME    READY   UP-TO-DATE   AVAILABLE   AGE
nginx   0/1     1            0           22s
nginx   1/1     1            1           23s
nginx   0/1     1            0           23s
nginx   1/1     1            1           23s
nginx   0/1     1            0           23s
nginx   1/1     1            1           23s
nginx   0/1     1            0           23s
nginx   1/1     1            1           23s
nginx   0/1     1            0           23s
nginx   1/1     1            1           23s
nginx   0/1     1            0           23s
nginx   1/1     1            1           24s
nginx   0/1     1            0           24s
nginx   1/1     1            1           24s
nginx   0/1     1            0           24s
nginx   1/1     1            1           24s
nginx   0/1     1            0           24s
nginx   1/1     1            1           24s
nginx   0/1     1            0           24s
nginx   1/1     1            1           24s
nginx   0/1     1            0           24s
nginx   1/1     1            1           25s
nginx   0/1     1            0           25s
nginx   1/1     1            1           25s
nginx   0/1     1            0           25s
nginx   1/1     1            1           25s
nginx   0/1     1            0           25s
```

**Comment by hansedong** [comment]
@mrlihanbo 

Sorry, it was my mistake. 

The problem was solved. I deployed kube-controller-manager additionally in the karmada cluster. I guess it conflicts with karmada-controller-manager. kube-controller-manager will set the nginx deploy status to 0/1, and karmada-controller-manager will set it to 1/1. They constantly want to achieve their expectations, resulting in 0/1->1 /1->0/1 This situation recurs. 

When I stopped kube-controller-manager, the problem was solved.
